cognito-post-authentication/src/test/java/no/unit/nva/cognito/service/CustomerApiClientTest.java

```diff
@@ -44,34 +45,52 @@ public void init() {
     }
 
     @Test
-    public void getCustomerIdReturnsCustomerIdOnInput() throws IOException, InterruptedException {
-        UUID uuid = UUID.randomUUID();
-        when(httpResponse.body()).thenReturn("{\"identifier\":\"" + uuid.toString() + "\"}");
+    public void getCustomerReturnsCustomerIdentifierOnInput() throws IOException, InterruptedException {
+        when(httpResponse.body()).thenReturn(customerResponseWithIdentifier());
         when(httpResponse.statusCode()).thenReturn(SC_OK);
         when(httpClient.send(any(), any())).thenReturn(httpResponse);
 
-        Optional<String> customerId = customerApiClient.getCustomerId(ORG_NUMBER);
+        Optional<CustomerResponse> customer = customerApiClient.getCustomer(ORG_NUMBER);
 
-        assertEquals(uuid.toString(), customerId.get());
+        assertEquals(SAMPLE_ID, customer.get().getIdentifier());
     }
 
     @Test
-    public void getCustomerIdReturnsEmptyOptionalOnInvalidJsonResponse() throws IOException, InterruptedException {
+    public void getCustomerReturnsCristinIdOnInput() throws IOException, InterruptedException {
+        when(httpResponse.body()).thenReturn(customerResponseWithCristinId());
+        when(httpResponse.statusCode()).thenReturn(SC_OK);
+        when(httpClient.send(any(), any())).thenReturn(httpResponse);
+
+        Optional<CustomerResponse> customer = customerApiClient.getCustomer(ORG_NUMBER);
+
+        assertEquals(SAMPLE_ID, customer.get().getCristinId());
+    }
+
+    @Test
+    public void getCustomerReturnsEmptyOptionalOnInvalidJsonResponse() throws IOException, InterruptedException {
         when(httpResponse.body()).thenReturn(GARBAGE_JSON);
         when(httpResponse.statusCode()).thenReturn(SC_OK);
         when(httpClient.send(any(), any())).thenReturn(httpResponse);
 
-        Optional<String> customerId = customerApiClient.getCustomerId(ORG_NUMBER);
+        Optional<CustomerResponse> customer = customerApiClient.getCustomer(ORG_NUMBER);
 
-        assertTrue(customerId.isEmpty());
+        assertTrue(customer.isEmpty());
     }
 
     @Test
-    public void getCustomerIdReturnsEmptyOptionalOnInvalidHttpResponse() throws IOException, InterruptedException {
+    public void getCustomerReturnsEmptyOptionalOnInvalidHttpResponse() throws IOException, InterruptedException {
         when(httpClient.send(any(), any())).thenThrow(IOException.class);
 
-        Optional<String> customerId = customerApiClient.getCustomerId(ORG_NUMBER);
+        Optional<CustomerResponse> customer = customerApiClient.getCustomer(ORG_NUMBER);
+
+        assertTrue(customer.isEmpty());
+    }
+
+    private String customerResponseWithIdentifier() {
+        return "{\"identifier\":\"" + SAMPLE_ID + "\"}";
```

Comment:
Maybe extract template "{\"%s\": \"%s\"}" for use with this and following method, passing constants in